openpype/hosts/max/plugins/publish/validate_renderpasses.py

```diff
@@ -0,0 +1,157 @@
+            The expected filename: {InstanceName}_RsCryptomatte.png
+
+        Args:
+            instance (pyblish.api.Instance): instance
+            filename (str): filename of the Max scene
+
+        Returns:
+            list: list of invalid filename which doesn't match
+                with the project name
+        """
+        invalid = []
+        file = rt.maxFileName
+        filename, ext = os.path.splitext(file)
+        if filename not in rt.rendOutputFilename:
+            cls.log.error(
+                "Render output folder "
+                "doesn't match the max scene name! "
+            )
+            invalid_folder_name = os.path.dirname(
+                rt.rendOutputFilename).replace(
+                    "\\", "/").split("/")[-1]
+            invalid.append(("Invalid Render Output Folder",
+                            invalid_folder_name))
+        beauty_fname = os.path.basename(rt.rendOutputFilename)
+        ext = os.path.splitext(beauty_fname)[-1].lstrip(".")
+        invalid_image_format = cls.get_invalid_image_format(
+            cls, instance, ext)
+        invalid.extend(invalid_image_format)
+        renderer = instance.data["renderer"]
+        if renderer in [
+            "ART_Renderer",
+            "Redshift_Renderer",
+            "V_Ray_6_Hotfix_3",
+            "V_Ray_GPU_6_Hotfix_3",
+            "Default_Scanline_Renderer",
+            "Quicksilver_Hardware_Renderer",
+        ]:
+            render_elem = rt.maxOps.GetCurRenderElementMgr()
+            render_elem_num = render_elem.NumRenderElements()
+            for i in range(render_elem_num):
+                renderlayer_name = render_elem.GetRenderElement(i)
+                renderpass = str(renderlayer_name).split(":")[-1]
+                rend_file = render_elem.GetRenderElementFilename(i)
+                if not rend_file:
+                    cls.log.error(
+                        f"No filepath for render element {renderpass}")
+                    invalid.append((f"Invalid {renderpass}",
+                                    "No filepath"))
+                rend_fname, ext = os.path.splitext(
+                    os.path.basename(rend_file))
+                if not rend_fname.lstrip(".").endswith(renderpass):
+                    cls.log.error(
+                        f"Filename for {renderpass} should "
+                        f"end with {renderpass}"
+                    )
+                    invalid.append((f"Invalid {renderpass}",
+                                    os.path.basename(rend_file)))
+                invalid_image_format = cls.get_invalid_image_format(
+                    cls, instance, ext)
+                invalid.extend(invalid_image_format)
+        elif renderer == "Arnold":
+            cls.log.debug(
+                "Renderpass validation not supported Arnold yet,"
+                " validation skipped...")
+
+        return invalid
+
+    def get_invalid_image_format(self, instance, ext):
```

Comment:
This function is called from a `classmethod` and thus should be a `classmethod` as well. Otherwise `self` here is actually referring to a class even though the code here looks like it would have access to an instance of the class.